awx/main/tests/unit/models/test_workflow_unit.py

```diff
@@ -194,18 +194,18 @@ def test_inherit_workflow_job_extra_vars(self, job_node_no_prompts):
             extra_vars={'a': 84}, **self.kwargs_base)
 
     def test_char_prompts_and_res_node_prompts(self, job_node_with_prompts):
+        # TBD: properly handle multicred credential assignment
         expect_kwargs = dict(
             inventory=job_node_with_prompts.inventory.pk,
-            credential=job_node_with_prompts.credential.pk,
```

Comment:
this is the exact kind of assertion that we're just not going to get with unit tests.

Reply:
Right :/

Reply:
I have to admit - I'm kind of doing the WFJT node stuff in parallel right now. This is entirely reasonable. New functionality around m2m relationships - new functional tests.

Reply:
Yea, unless/until Django gives us some mechanisms for dealing with this more simply, I think the reality of the situation is that we'll really want functional tests for this sort of thing.  Otherwise it's mock hell.